pkg/flare/flare.go

```diff
@@ -0,0 +1,435 @@
+// Processed files are read line-by-line. Each line is filtered according to the date range,
+// sanitized from secrets, and is written to the file in outputPath.
+func (f *Flare) ProcessLogFiles(inputPath, outputPath string, startDate, endDate *time.Time) (retErr error) {
+	isDir, err := isDirectory(inputPath)
+	if err != nil {
+		return err
+	}
+
+	if !isDir {
+		return f.processLogFile(inputPath, outputPath, startDate, endDate)
+	} else {
+		files, err := os.ReadDir(inputPath)
+		if err != nil {
+			return err
+		}
+
+		for _, file := range files {
+			if file.IsDir() {
+				continue
+			}
+			err = f.processLogFile(file.Name(), outputPath, startDate, endDate)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func extractDateFromPlainTextLine(line, logDateLayout string) (time.Time, error) {
+	idxs := plainTextLogDateRegex.FindIndex([]byte(line))
+	if idxs == nil {
+		return time.Time{}, ErrDateNotFound
+	}
+
+	strDate := line[idxs[0]+1 : idxs[1]-1]
+	date, err := time.Parse(logDateLayout, strDate)
+	if err != nil {
+		return time.Time{}, err
+	}
+	return date, nil
+}
+
+func extractDateFromJSONLine(line, logDateLayout string) (time.Time, error) {
+	logPropsMap := make(map[string](interface{}))
+	err := json.Unmarshal([]byte(line), &logPropsMap)
+	if err != nil {
+		return time.Time{}, err
+	}
+	lineDateStr, ok := logPropsMap["time"].(string)
+	if !ok {
+		return time.Time{}, ErrExtractDateFromJSONLogLine
+	}
+	lineDate, err := time.Parse(logDateLayout, lineDateStr)
+	if err != nil {
+		return time.Time{}, err
+	}
+	return lineDate, nil
+}
+
+func (f *Flare) handleLogLine(line string, start, end *time.Time) (string, error) {
+	lineDate, err := dateExpractors[f.logFormat](line, f.LogDateLayout)
+	if err != nil {
+		return "", err
+	}
+	if (start == nil || lineDate.Equal(*start) || lineDate.After(*start)) &&
+		(end == nil || lineDate.Equal(*end) || lineDate.Before(*end)) {
+		redactedLine, err := f.redactSecrets(line)
+		if err != nil {
+			return "", err
+		}
+		return redactedLine, nil
+	}
+	return "", nil
+}
+
+func (f *Flare) processLogFile(inputFileName, outputPath string, startDate, endDate *time.Time) (retErr error) {
+	skippedLines := 0
+	sourceFile, err := os.Open(inputFileName)
+	if err != nil {
+		return err
+	}
+	defer sourceFile.Close()
+
+	outputFileName := fmt.Sprintf("%s%s", outputPath, inputFileName)
+	outputFile, err := os.OpenFile(outputFileName, os.O_WRONLY|os.O_CREATE, FilePremissions)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if e := outputFile.Close(); retErr == nil {
+			retErr = e
+		}
+	}()
+
+	bw := bufio.NewWriter(outputFile)
+	defer func() {
+		if e := bw.Flush(); retErr == nil {
+			retErr = e
+		}
+	}()
+
+	first := true
+	scanner := bufio.NewScanner(sourceFile)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		pLine, err := f.handleLogLine(line, startDate, endDate)
+		if err != nil {
+			skippedLines += 1
+			fmt.Println("failed to process log line: ", err)
```

Comment:
Please report errors to os.Stderr.  This will happen directly if you use Warning().

Reply:
`Warning()` is in `lakectl`. Replaced `Println` with `Fprintf`